billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/export/webservice/Client.java

```diff
@@ -0,0 +1,497 @@
+            } catch (DatatypeConfigurationException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        return null;
+    }
+
+    private DebitCreditIndicator getDebitCreditIndicator(final GenericInvoiceEntry entry) {
+        final DebitCreditIndicator debitCreditIndicator;
+        switch (entry.getCreditOrDebit()) {
+            case DEBIT:
+                debitCreditIndicator = DebitCreditIndicator.D;
+                break;
+            case CREDIT:
+                debitCreditIndicator = DebitCreditIndicator.C;
+                break;
+            default:
+                throw new RuntimeException("unknow value for: " + entry.getCreditOrDebit());
+        }
+        return debitCreditIndicator;
+    }
+
+    private String getReference(final GenericInvoiceEntry entry) {
+        String reference = null;
+        if (entry instanceof PTCreditNoteEntryEntity) {
+            reference = ((PTCreditNoteEntryEntity) entry).getReference().getNumber();
+        }
+        return reference;
+    }
+
+    private String getRegionCodeFromISOCode(String regionCode) {
+        if (regionCode.equals("PT-20")) {
+            return "PT-AC";
+        }
+        if (regionCode.equals("PT-30")) {
+            return "PT-MA";
+        }
+        return "PT";
+    }
+
+    private TaxType getTaxType(PTTaxEntity entity) {
+        switch (entity.getTaxRateType()) {
+            case PERCENTAGE:
+                return TaxType.IVA;
+            case FLAT:
+                return TaxType.IS;
+            case NONE:
+                return TaxType.NS;
+            default:
+                throw new RuntimeException(entity.getTaxRateType().toString());
+        }
+    }
+
+    private InvoiceTypeType processInvoiceType(PTGenericInvoice ptGenericInvoice) {
+        switch (ptGenericInvoice.getType()) {
+            case FT:
+                return InvoiceTypeType.FT;
+            case FS:
+                return InvoiceTypeType.FS;
+            case FR:
+                return InvoiceTypeType.FR;
+            case NC:
+                return InvoiceTypeType.NC;
+            case ND:
+                return InvoiceTypeType.ND;
+            default:
+                throw new RuntimeException(ptGenericInvoice.getType().toString());
+        }
+    }
+
+    private InvoiceStatus processStatus(PTGenericInvoice ptGenericInvoice) throws DatatypeConfigurationException {
+        final InvoiceStatus invoiceStatus = new InvoiceStatus();
+        if (ptGenericInvoice.isCancelled()) {
+            invoiceStatus.setInvoiceStatus("A");
+        } else if (ptGenericInvoice.isBilled()) {
+            invoiceStatus.setInvoiceStatus("F");
+        } else if (ptGenericInvoice.isSelfBilled()) {
+            invoiceStatus.setInvoiceStatus("S");
+        } else {
+            invoiceStatus.setInvoiceStatus("N");
+        }
+        invoiceStatus.setInvoiceStatusDate(formatDateTime(ptGenericInvoice.getDate()));
```

Comment:
This date doesn't account for the changes in upstream that added timezones to businesses and local dates to SAFT and export logics.

Reply:
updated